packages/dds/tree/src/test/forestTestSuite.ts

```diff
@@ -0,0 +1,724 @@
+            assert.deepStrictEqual(parentPath2, expectedParent);
+
+            const expectedChild1: UpPath = {
+                parent: expectedParent,
+                parentField: brand("data"),
+                parentIndex: 0,
+            };
+
+            const expectedChild2: UpPath = {
+                parent: expectedParent,
+                parentField: brand("data"),
+                parentIndex: 1,
+            };
+
+            assert.deepStrictEqual(childPath1, expectedChild1);
+            assert.deepStrictEqual(childPath2, expectedChild2);
+
+            assert.equal(forest.tryMoveCursorToNode(parentAnchor, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, undefined);
+            assert.equal(forest.tryMoveCursorToNode(childAnchor1, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, 1);
+            assert.equal(forest.tryMoveCursorToNode(childAnchor2, cursor), TreeNavigationResult.Ok);
+            assert.equal(cursor.value, 2);
+
+            // Cleanup is not required for this test (since anchor set will go out of scope here),
+            // But make sure it works:
+            forest.anchors.forget(parentAnchor);
+            forest.anchors.forget(childAnchor1);
+            forest.anchors.forget(childAnchor2);
+            forest.anchors.forget(parentAnchor2);
+            assert(forest.anchors.isEmpty());
+        });
+
+        it("using an anchor that went away returns NotFound", () => {
+            const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+            const dependent = new MockDependent("dependent");
+            recordDependency(dependent, forest);
+
+            const content: JsonableTree[] = [
+                {
+                    type: jsonObject.name,
+                    fields: {
+                        data: [
+                            { type: jsonNumber.name, value: 1 },
+                            { type: jsonNumber.name, value: 2 },
+                        ],
+                    },
+                },
+            ];
+            initializeForest(forest, content.map(singleTextCursor));
+
+            const cursor = forest.allocateCursor();
+            moveToDetachedField(forest, cursor);
+            assert(cursor.firstNode());
+            cursor.enterField(brand("data"));
+            cursor.enterNode(0);
+            const firstNodeAnchor = cursor.buildAnchor();
+            cursor.clear();
+
+            const mark: Delta.Delete = { type: Delta.MarkType.Delete, count: 1 };
+            const delta: Delta.Root = new Map([[rootFieldKeySymbol, [0, mark]]]);
+            // TODO: make type-safe
+            forest.applyDelta(delta);
+
+            assert.equal(
+                forest.tryMoveCursorToNode(firstNodeAnchor, cursor),
+                TreeNavigationResult.NotFound,
+            );
+        });
+
+        describe.only("can clone", () => {
```

Comment:
It would be good to add a test to verify that editing a cloned forest does not change the original and does cause invalidation like the editing the original would.

Reply:
Incidentally, this may force a change to the `clone` interface so that it returns an editable forest, which seems like an improvement.